fastplotlib/graphics/selectors/_rectangle.py

```diff
@@ -0,0 +1,511 @@
+                                        np.array([], dtype=np.float32).reshape(0, 3)
+                                    )
+                                else:
+                                    data_selections.append(g.data[s])
+                return data_selections
+            else:
+                if ixs.size == 0:
+                    # empty selection
+                    return np.array([], dtype=np.float32).reshape(0, 3)
+
+                s = slice(
+                    ixs[0], ixs[-1] + 1
+                )  # add 1 to end because these are direct indices
+                # slices n_datapoints dim
+                # slice with min, max is faster than using all the indices
+
+                # get missing ixs
+                missing_ixs = np.setdiff1d(np.arange(ixs[0], ixs[-1] + 1), ixs) - ixs[0]
+
+                match mode:
+                    case "full":
+                        return source.data[s]
+                    case "partial":
+                        if len(missing_ixs) > 0:
+                            data = source.data[s].copy()
+                            data[missing_ixs] = np.nan
+                            return data
+                        else:
+                            return source.data[s]
+                    case "ignore":
+                        if len(missing_ixs) > 0:
+                            warnings.warn(
+                                "You have selected 'ignore' mode. Selected graphic has incomplete indices. "
+                                "Move the selector or change the mode to one of `partial` or `full`."
+                            )
+                            return np.array([], dtype=np.float32)
+                        else:
+                            return source.data[s]
+
+    def get_selected_indices(
+        self, graphic: Graphic = None
+    ) -> Union[np.ndarray, List[np.ndarray]]:
+        """
+        Returns the indices of the ``Graphic`` data bounded by the current selection.
+
+        These are the data indices which correspond to the data under the selector.
+
+        Parameters
+        ----------
+        graphic: Graphic, default ``None``
+            If provided, returns the selection indices from this graphic instrad of the graphic set as ``parent``
+
+        Returns
+        -------
+        Union[np.ndarray, List[np.ndarray]]
+            data indicies of the selection, list of np.ndarray if the graphic is a collection
+        """
+        # get indices from source
+        source = self._get_source(graphic)
+
+        # selector (xmin, xmax, ymin, ymax) values
+        bounds = self.selection
+
+        # image data does not need to check for mode because the selector is always bounded
+        # to the image
+        if "Image" in source.__class__.__name__:
+            ys = np.arange(bounds[0], bounds[1], dtype=int)
+            xs = np.arange(bounds[2], bounds[3], dtype=int)
+            return [xs, ys]
```

Comment:
```suggestion
            xs = np.arange(bounds[0], bounds[1], dtype=int)
            ys = np.arange(bounds[2], bounds[3], dtype=int)
            return [xs, ys]
```

Image data is [row, cols], same as [y-indices, x-indices]